backend/dataall/modules/dataset_sharing/services/data_sharing_service.py

```diff
@@ -176,6 +177,15 @@ def approve_share(cls, engine: Engine, share_uri: str) -> bool:
                 new_share_state = share_sm.run_transition(ShareObjectActions.Finish.value)
                 share_sm.update_state(session, share, new_share_state)
 
+                if share_successful:
```

Comment:
I think we still run into issues here in the following scenario:

- I approve share for 1 table and 1 folder
- 1 table succeeds but 1 folder fails (no permissions attached to requestor)
- I delete a re-submit folder and it succeeds (requestor gets permissions only to folder NOT to table)

May need to attach permissions 1 by 1 to each item to avoid the above

Reply:
done